components/script/dom/document.rs

```diff
@@ -1975,6 +1978,16 @@ impl Document {
                            ReflowQueryType::NoQuery,
                            ReflowReason::ElementStateChanged);
     }
+
+    pub fn serialize(&self) -> Fallible<DOMString> {
```

Comment:
Putting it inline in `xmlhttprequest.rs`? There's no need for a function altogether.